site/docs/releases.md

```diff
@@ -71,6 +71,35 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
   ...
 </dependencies>
 ```
+
+### 1.6.0 release
+Apache Iceberg 1.6.0 was released on July 23, 2024.
+
+The 1.6.0 release contains fixes and dependency updates.
+
+* Build
+    - Upgrade to Gradle 8.9 ([\#10686](https://github.com/apache/iceberg/pull/10686))
+    - Move to goooler shadow plugin ([\#10568](https://github.com/apache/iceberg/pull/10568))
+* Core
+    - Fix NPE during conflict handling of NULL partitions ([\#10680](https://github.com/apache/iceberg/pull/10680))
+    - Fix ParallelIterable memory leak where queue continues to be populated even after iterator close ([\#9402](https://github.com/apache/iceberg/pull/9402))
+    - REST Catalog: deprecate oauth/tokens endpoint ([\#10603](https://github.com/apache/iceberg/pull/10603))
+    - REST Catalog: fix create v1 table ([\#10369](https://github.com/apache/iceberg/pull/10369))
+    - REST Catalog: disallow overriding "credential" in table sessions ([\#10345](https://github.com/apache/iceberg/pull/10345))
+    - REST Catalog: assume issued_token_type is access_token to fully comply with RFC 6749 )[\#10314](https://github.com/apache/iceberg/pull/10314))
+    - JDBC Catalog: exclude unexpected namespaces ([\#10498](https://github.com/apache/iceberg/pull/10498))
+* Spark
+    - Writes/actions should only perform cleanup if failure is cleanable ([\#10373](https://github.com/apache/iceberg/pull/10373))
+    - Fix issue when partitioning by UUID ([\#8250](https://github.com/apache/iceberg/pull/8250))
+* Kafka Connect
+    - Add commit coordinator ([\#10351](https://github.com/apache/iceberg/pull/10351))
+* Azure
+    - Make AzureProperties with shared-key credentials serializable ([\#10045](https://github.com/apache/iceberg/pull/10045))
+
+For more details, please visit [Github](https://github.com/apache/iceberg/releases/tag/apache-iceberg-1.6.0).
```

Comment:
Gonna tackle that.

Generally speaking, I think the releases page should be a bit revisited and automatic, else it's not easy to update.

Reply:
I checked and multi-engine-support doc is already up to date.

Reply:
> I checked and multi-engine-support doc is already up to date.

Ack.

Reply:
I usually used to prepare release notes for non-patch release from community sync highlights since previous release. Thats how I did for 1.5.0, It covers most of the major things. I know it is quite a work. But without this it is hard to find the highlights of the release in the pool of few hundred/thousand commits for the user.